docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,162 @@
+- Ensuring the transaction satisfies the endorsement policies of the namespaces of the corresponding chaincodes.
+
+
+## Approach
+
+Restrict Fabric functionality:
+
+- keep Fabric validation changes in sync with TLCC
+- Reduces validation logic to a minimum
+    - minimizes catch-up 
+- Restrict the notion of endorsement policy; 
+- Simplify chaincode versioning (TODO)
+    - No chaincode updates for now
+- Support for default MSP only
+
+
+Any transactions/blocks with unsupported features are ignored/aborted
+
+Defined process to develop TLCC:
+- keep redundant validation code (in stock peer and TLCC) in sync
+- keep code relation traceable and changes trackable
+- consistent naming and code structure as much as possible
+- bonus: 
+    - automated code-changes notification; notifies whenever relevant Fabric code changes and might FPC must be updates.
+
+### Non-features
+
+- No state-based endorsement
+- No custom MSP (no idemix)
+- No support for custom endorsing/validation plugins for non-FPC and FPC chaincodes
+- Authentication and decorators
+- TODO check for more non-features in RFC
+
+## Design
+
+### Chaincode execution support
+    - secure channel
+    - validate proposal
+        - check org is "write"
+        - replay protection
+
+### Ledger maintainance
+
+Channel configuration:
+- Parse channel definition
+    - parse consensus definition
+- parse channel reader/writers
+- validate block signatures (Note that this check implies correctness as onfig blocks are validated by the ordering service ([see code](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/orderer/common/msgprocessor/standardchannel.go#L131)))
+- Maintain msp metadata for signature validation
+- parse Lifecycle policy 
+    - validate Lifecycle policy
+    We only support: a majority of Org.member (See [Link](https://hyperledger-fabric.readthedocs.io/en/release-2.2/chaincode_lifecycle.html#install-and-define-a-chaincode))
+
+Access control:
+- delegate this "reader/writer/admin" check to admins;
+- TLCC only verifies endorsement policies and thereby implicitly the "writers" check is performed by the endorsers
+
+MSP:
+- One Org - One MSP mapping - One root cert
+- Implement X509-based MSP
+    - Restrict:
+        - no intermediate
+        - no CRLs
+        - only support member role
+    - any certificate will match to role member
+
+Endorsing:
+- Phase1: Designated enclave only
+- Phase2: Any enclave that runs a particular FPC chaincode
+
+Versioning:
+- single autonomously monotonously increasing version number??
+
+Non-FPC Validation:
+- Transaction submitter identity validation
+    - submitter satisfies channel's writes policy
+- ERCC endorsement signatures verification
+- ERCC endorsement policy validation
+    - we only support (explicityly) Majority endorsement policy
+    - Restrict to ERCC namespace only
+
+FPC Validation:
+- Introduce FPC transaction type (similar to introduce FPC namespace) and create
```

Comment:
seems to me we should split this into 2: in the current design we essentially define the type somewhat implicitly by the presence of the fpc custom validation plugin.  Do we really want to tackle changes of fabric core in phase 1?  If it is possible with minimal changes it certainly could have various advantages but it could also be a rathole of changes also on client side etc?